components/apps_form/apps_form_field/apps_form_select_field.tsx

```diff
@@ -73,6 +88,21 @@ export default class AppsFormSelectField extends React.PureComponent<Props, Stat
         return this.props.performLookup(this.props.field.name, userInput);
     }
 
+    loadDynamicUserOptions = async (userInput: string): Promise<AppSelectOption[]> => {
+        const usersSearchResults: UserAutocomplete = await this.props.actions.autocompleteUsers(userInput.toLowerCase());
+        return usersSearchResults.users.map((user) => ({...user, label: this.props.teammateNameDisplay ? displayUsername(user, this.props.teammateNameDisplay) : user.username, value: user.id, icon_data: Client4.getUsersRoute() + '/' + user.id + '/image?_=' + (user.last_picture_update || 0)}));
+    }
+
+    loadDynamicChannelOptions = async (userInput: string): Promise<AppSelectOption[]> => {
+        let channelsSearchResults: Channel[] = [];
+
+        await this.props.actions.autocompleteChannels(userInput.toLowerCase(), (data) => {
+            channelsSearchResults = data;
+        }, () => {});
+
+        return channelsSearchResults.map((channel) => ({...channel, label: channel.name, value: channel.id}));
```

Comment:
Can we use a "globe" or "padlock" whether the channel is public or private? This can be done with the `CompassIcon` component

Reply:
I'm fine with deferring this item